lms/djangoapps/instructor_task/api.py

```diff
@@ -446,7 +446,9 @@ def generate_certificates_for_students(request, course_key, students=None):  # p
     """
     if students:
         task_type = 'generate_certificates_certain_student'
-        students = [student.id for student in students]
+        # If students is not string then traverse the students object.
+        if not isinstance(students, str):
```

Comment:
The only places where this is called are:
1) https://github.com/edx/edx-platform/blob/master/lms/djangoapps/instructor/views/api.py#L3278-L3279
2) https://github.com/edx/edx-platform/blob/master/lms/djangoapps/certificates/views/support.py#L263-L264

In both of those cases, certificates are only being generated for one student. With that in mind, I'd suggest updating [`generate_certificates_for_students`](https://github.com/edx/edx-platform/blob/master/lms/djangoapps/instructor_task/api.py#L440)'s params to:
```python
def generate_certificates_for_students(request, course_key, student_set, specific_student):
    pass
```
The idea here is that `generate_certificates_for_students` would either take a set of students (a string like `"whitelisted_not_generated"`) or a specific student (either a user id or user object). Then we don't have to do this crazy type checking, and it's more semantic.